src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1210,6 +1210,7 @@ def __init__(self,
                  extra_container_args: Optional[GeneralArgList] = None,
                  extra_entrypoint_args: Optional[GeneralArgList] = None,
                  custom_configs: Optional[List[CustomConfig]] = None,
+                 health_check_interval: Optional[int] = None,
```

Comment:
Is health_check_interval meant to be an int or a str? Here it's int but elsewhere it looks more like a string ("2s" for instance). If it can be either, let's make sure that's represented in the type annotations too. If it can be a str let's also add some validatation to the the string value (maybe something like the regex `[0-9]+[msh]?` )

Reply:
I found the list of suffixes the `inter` field for haproxy supports: https://docs.haproxy.org/2.4/configuration.html#5.2-inter
This can be used to prevalidate the input before putting it in the haproxy config file.